### ISSUE TITLE
Sequentially editing `@link` content, running go-to-definition, then diagnostics, incorrectly creates new errors

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#50341]
```ts
declare class Stuff {
    /** For more thorough tests, use {@link checkBarIs} */
    checkFooLengthIs(len: number): void;

    checkFooIs(value: object): void;
}
```

If you quickly replace `checkBarIs` with `checkFooIs` in the above example, then VS Code will sometimes make a `definitionAndBoundSpan` request. That request will get served, and then the editor will request errors. The errors returned then become nonsensical.

**Expected**: No errors
**Actual**: 

* Line 2:45-55 - "Cannot find name 'checkFooIs'. Did you mean the instance member 'this.checkFooIs'?"
* Line 3:5-21 - "Cannot find name 'checkFooIs'."

_____

<details>
<summary>Here's a GIF of this in action</summary>

![jsdocFastTypingInLinkTag](https://user-images.githubusercontent.com/972891/185224036-2b94e25d-0d42-40af-8c4c-bf631178ec44.gif)
</details>

_____

Here's a relevant fourslash test that currently fails.

```ts
/// <reference path="./fourslash.ts" />

////declare class Stuff {
////    /** For more thorough tests, use {@link [|checkFooIs|]} */
////    checkFooLengthIs(len: number): void;
////
////    /*checkFooIsDef*/checkFooIs(value: object): void;
////}

const range = test.ranges()[0];
goTo.selectRange(range);
verify.goToDefinitionIs("checkFooIsDef");
verify.noErrors();
```

**Comment by DanielRosenwasser** [comment]
Unfortunately this isn't related to the `@link` tag - I think it occurs in any JSDoc.

```
/**
 * @param asd {[|UnresolvedTypeReference|]}
 */
function blah(asd) {
}
```

If you do the same sort thing - run a semantic operation on `UnresolvedTypeReference`, and *then* request errors - then you'll start seeing errors in the JSDoc, even in a `.ts` file.

**Comment by DanielRosenwasser** [comment]
#49109 seems strongly related.

**Comment by DanielRosenwasser** [comment]
Another

```ts
/**
 * @type {{ x: $ }}
 */
```

Type in `this[""]` and you'll get the error:

```
A 'this' type is available only in a non-static member of a class or interface.
```
